src/Domain/CampaignRepository.php

```diff
@@ -28,26 +28,26 @@ class CampaignRepository extends SalesforceReadProxyRepository
      */
     public function findRecentLiveAndPendingGiftAidApproval(): array
     {
-        $shortLookBackDate = (new DateTime('now'))->sub(new \DateInterval('P7D'));
-        $extendedLookBackDate = (new DateTime('now'))->sub(new \DateInterval('P9M'));
-
-        $qb = $this->getEntityManager()->createQueryBuilder()
-            ->select('c')
-            ->from(Campaign::class, 'c')
-            ->innerJoin('c.charity', 'charity')
-            ->where('c.endDate >= :shortLookBackDate')
-            ->orWhere(<<<EOT
+        // make this include ones where charity needs update
+        $query = $this->getEntityManager()->createQuery(<<<DQL
+            SELECT c FROM MatchBot\Domain\Campaign c
+            INNER JOIN c.charity charity
+            WHERE c.endDate >= :shortLookBackDate OR (
                 charity.tbgClaimingGiftAid = 1 AND
                 charity.tbgApprovedToClaimGiftAid = 0 AND
                 c.endDate >= :extendedLookbackDate
-EOT
             )
-            ->orderBy('c.createdAt', 'ASC')
-            ->setParameter('shortLookBackDate', $shortLookBackDate)
-            ->setParameter('extendedLookbackDate', $extendedLookBackDate);
+            ORDER BY c.createdAt ASC
+            DQL
```

Comment:
Changed this from query builder to DQL for easier editing, then edited, then thought better and put back to how it was before. I think the change to DQL is worth keeping though.